proposals/NNNN-shader-execution-reordering.md

```diff
@@ -0,0 +1,1388 @@
+    float,            ; tMax
+    PayloadT*)        ; payload
+    nounwind
+```
+
+#### HitObject_FromRayQuery
+
+```DXIL
+declare %dx.types.HitObject @dx.op.hitObject_FromRayQuery(
+    i32,                           ; opcode
+    %dx.types.RayQueryHandle)      ; ray query
+    nounwind argmemonly
+
+```
+This is used for the HLSL overload of `HitObject::FromRayQuery` that only takes `RayQuery`.
+
+#### HitObject_FromRayQueryWithAttrs
+
+```DXIL
+declare %dx.types.HitObject @dx.op.hitObject_FromRayQuery.AttrT(
+    i32,                           ; opcode
+    %dx.types.RayQueryHandle,      ; ray query
+    AttrT*)                        ; attributes
+    nounwind argmemonly
+```
+This is used for the HLSL overload of `HitObject::FromRayQuery` that takes `RayQuery` and the user-defined `Attribute` struct.
+`AttrT` is the user-defined intersection attribute struct type. See `ReportHit` for definition.
+
+#### HitObject_MakeMiss
+
+```DXIL
+declare %dx.types.HitObject @dx.op.hitObject_MakeMiss(
+    i32,                           ; opcode
+    i32,                           ; ray flags
+    i32,                           ; miss shader index
+    float,                         ; ray origin x
+    float,                         ; ray origin y
+    float,                         ; ray origin z
+    float,                         : tMin
+    float,                         ; ray direction x
+    float,                         ; ray direction y
+    float,                         ; ray direction z
+    float)                         : tMax
+    nounwind readnone
+```
+
+#### HitObject_MakeNop
+
+```DXIL
+declare %dx.types.HitObject @dx.op.hitObject_MakeNop(
+    i32)                           ; opcode
+    nounwind readnone
+```
+
+#### HitObject_Invoke
+
+```DXIL
+declare void @dx.op.hitObject_Invoke.PayloadT(
+    i32,                           ; opcode
+    %dx.types.HitObject,           ; hit object
+    PayloadT*)                     ; payload
+    nounwind
+```
+
+#### ReorderThread
+
+Operation that reorders the current thread based on the supplied hints and
+`HitObject`. The canonical lowering of the
+HLSL intrinsic `ReorderThread( uint CoherenceHint, uint NumCoherenceHintBitsFromLSB )`
+uses `undef` for the `HitObject` parameter.
+
+```DXIL
+declare void @dx.op.reorderThread(
+    i32,                      ; opcode
+    %dx.types.HitObject,      ; hit object
+    i32,                      ; coherence hint
+    i32)                      ; num coherence hint bits from LSB
+    nounwind
+```
+
+#### Generic State Value getters
+
+State value getters return scalar, vector or matrix values dependent on the provided opcode.
+Scalar getters use the `hitobject_StateScalar` dxil intrinsic and the return type is the state value type.
+Vector getters use the `hitobject_StateVector` dxil intrinsic and the return type is the vector element type.
+Matrix getters use the `hitobject_StateMatrix` dxil intrinsic and the return type is the matrix element type.
```

Comment:
I think it might be clearer to leave these out of the table above and give them their own special section. It already looks like a new topic because of the table. I think dropping them to the bottom instead of in the middle of the list of intrinsics would be less confusing.

Reply:
I moved them last, but I think I disagree about leaving them out of the initial opcode table. That table is supposed to list all opcodes and their values. The later table doesn't have the values.